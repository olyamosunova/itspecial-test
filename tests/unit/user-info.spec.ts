import { mount, createLocalVue } from "@vue/test-utils";
import Vuex from "vuex";
import UserInfo from "@/components/user-info/User-info.vue";

const localVue = createLocalVue();
localVue.use(Vuex);

const chosenUser = {
    id: 261,
    firstName: "Meribeth",
    lastName: "Haglund",
    email: "VDaly@odio.io",
    phone: "(411)804-5726",
    adress: {
        streetAddress: "9749 Ante Ct",
        city: "El Dorado Hills",
        state: "NV",
        zip: "71328"
    },
    description: "aliquam lacus odio magna lacus sed sed lacus sed convallis ipsum sollicitudin libero magna hendrerit tortor magna porta orci sed aliquam sed vitae magna egestas placerat curabitur amet suspendisse lacus sagittis sollicitudin"
};

describe('unit tests for UserInfo component', () => {
    let getters;
    let store: any;

    beforeEach(() => {
        getters = {
            chosenUser: () => chosenUser,
        };

        store = new Vuex.Store({
            getters
        });
    });

    it('should match the snapshot', () => {
        const wrapper = mount(UserInfo, { store, localVue });
        expect(wrapper.html()).toMatchSnapshot();
    })
});
