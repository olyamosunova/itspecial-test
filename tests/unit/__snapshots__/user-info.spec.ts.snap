// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unit tests for UserInfo component should match the snapshot 1`] = `
<div class="card">
  <div class="card__body">
    <p class="card__text">Selected user: <b>Meribeth Haglund</b></p>
    <p class="card__text"> Description:
      <textarea disabled="disabled" class="card__textarea" style="height: 2px;"></textarea>
    </p>
    <p class="card__text">Residence address: <b>9749 Ante Ct</b></p>
    <p class="card__text">City: <b>El Dorado Hills</b></p>
    <p class="card__text">Province / State: <b>NV</b></p>
    <p class="card__text">Zip: <b>71328</b></p>
  </div>
</div>
`;
